### PR TITLE
compile e2e perf tests with move2

### PR DESCRIPTION
**Pull request by igor-aptos**
## Description
We have a suite of move code that we use for perf e2e tests (where we publish those modules, and execute transactions against them)

Enable move2 compiler for those - to allow for testing perf of new features, as well as testing move2 compiler more.
We can see if perf changes for any of the tests - wouldn't expect it does.

## How Has This Been Tested?
single node perf benchmark

## Type of Change
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [x] Validator Node
- [x] Move/Aptos Virtual Machine

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by wrwg — APPROVED** [review]

File: testsuite/module-publish/src/main.rs [hunk 1]
```diff
@@ -148,7 +148,7 @@ use std::collections::HashMap;"#,
 fn write_package(file: &mut File, package_path: PathBuf, package_name: &str) -> String {
     println!("Building package {}", package_name);
     // build package
-    let package = BuiltPackage::build(package_path, BuildOptions::default())
+    let package = BuiltPackage::build(package_path, BuildOptions::move_2())
```
Comment (georgemitenkov) [inline testsuite/module-publish/src/main.rs:151]: Can we specify optimizer options as well? Not sure if those are enabled by default, cc @vineethk

File: aptos-move/e2e-benchmark/src/main.rs [hunk 2]
```diff
@@ -117,10 +117,10 @@ fn main() {
             num_objects: 100,
             object_payload_size: 10 * 1024,
         }),
-        (66, EntryPoints::InitializeVectorPicture { length: 40 }),
+        (61, EntryPoints::InitializeVectorPicture { length: 40 }),
         (16, EntryPoints::VectorPicture { length: 40 }),
         (16, EntryPoints::VectorPictureRead { length: 40 }),
-        (28299, EntryPoints::InitializeVectorPicture {
+        (23256, EntryPoints::InitializeVectorPicture {
```
Comment (georgemitenkov) [inline aptos-move/e2e-benchmark/src/main.rs:123]: Do we need to update executor-benchmark numbers as well? Seems like a regression

**Review by georgemitenkov — APPROVED** [review]

File: testsuite/module-publish/src/main.rs (hunk 1 above)
Comment (vineethk) [reply testsuite/module-publish/src/main.rs:151]: There are a set of optimizations enabled by default.

Comment (georgemitenkov) [reply testsuite/module-publish/src/main.rs:151]: Great!

File: aptos-move/e2e-benchmark/src/main.rs (hunk 2 above)
Comment (igor-aptos) [reply aptos-move/e2e-benchmark/src/main.rs:123]: this is a perf improvement, as this is in microseconds

Comment (georgemitenkov) [reply aptos-move/e2e-benchmark/src/main.rs:123]: Ah, awesome then

**Pull request merged**